File: solutions/3189-find-champion-ii/find-champion-ii.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        can_win = [True] * n
        for a, b in edges:
            can_win[b] = False
        winner = -1
        winner_count = 0
        for i in range(0, n):
            if can_win[i]:
                winner = i
                winner_count += 1
        if winner_count == 1:
            return winner
        return -1

File: solutions/3189-find-champion-ii/find-champion-ii.ts
// TypeScript

function findChampion(n: number, edges: number[][]): number {
    const isUndefeated: boolean[] = new Array(n).fill(true);
    
    for (const [winner, loser] of edges) {
        isUndefeated[loser] = false;
    }
    
    let champion: number = -1;
    let championCount: number = 0;
    
    for (let team = 0; team < n; team++) {
        if (isUndefeated[team]) {
            champion = team;
            championCount++;
        }
    }
    
    return championCount === 1 ? champion : -1;
}